arrow-cast/src/display.rs

```diff
@@ -169,10 +217,161 @@ impl<'a> FormatOptions<'a> {
         Self { types_info, ..self }
     }
 
-    /// Returns true if type info should be included in visual representation of batches
+    /// Overrides the [`ArrayFormatterFactory`] used to instantiate custom [`ArrayFormatter`]s.
+    pub const fn with_formatter_factory(
+        self,
+        formatter_factory: &'a dyn ArrayFormatterFactory,
+    ) -> Self {
```

Comment:
I think it is more common in the arrow APIs to pass the Option instead of having both a with and without method


```suggestion
    /// Overrides the [`ArrayFormatterFactory`] used to instantiate custom [`ArrayFormatter`]s.
    pub const fn with_formatter_factory(
        self,
        formatter_factory: Option<&'a dyn ArrayFormatterFactory>,
    ) -> Self {
```

Then you can remove the other method